### PR TITLE
change fingerprint collection to hashmap

### PR DESCRIPTION
**Pull request by tarkah**
for #108

### CODE REVIEW

File: src/parse.rs [hunk 1]
```diff
@@ -97,7 +103,7 @@ async fn file_parsing_regex() -> Result<ParsingPatterns> {
 pub async fn read_addon_directory<P: AsRef<Path>>(
     fingerprint_collection: Arc<Mutex<Option<FingerprintCollection>>>,
     root_dir: P,
-    flavor: &Flavor,
+    flavor: Flavor,
```
Comment (tarkah) [inline src/parse.rs:106]: Things that are `Copy` should just get passed as owned, not a reference

File: src/parse.rs [hunk 2]
```diff
@@ -137,17 +143,15 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
...
+    let new_fingerprints: Vec<_> = all_dirs
         .par_iter() // Easy parallelization
         .map(|dir_name| {
             let addon_dir = root_dir.join(dir_name);
             // If we have a stored fingerprint on disk, we use that.
-            if let Some(fingerprint) = fingerprint_collection
-                .iter()
-                .find(|f| &f.flavor == flavor && &f.title == dir_name)
-            {
+            if let Some(fingerprint) = fingerprints.iter().find(|f| &f.title == dir_name) {
```
Comment (tarkah) [inline src/parse.rs:154]: We no longer need to filter on flavor since this data set is now specific for the flavor

File: src/parse.rs [hunk 3]
```diff
@@ -159,36 +163,22 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
...
         .filter(|f| f.hash.is_some())
         .collect();
 
-    // Get the other flavors, we want saved as well.
-    let other_flavors_fingerprints: Vec<_> = fingerprint_collection
-        .clone()
-        .into_iter()
-        .filter(|f| &f.flavor != flavor)
-        .collect();
-
```
Comment (tarkah) [inline src/parse.rs:176]: Yay, we can get rid of this now!

Comment (casperstorm) [reply src/parse.rs:176]: **AWESOME**.

**Comment by casperstorm** [comment]
This is great. Its tested and is working flawless as always.

**Pull request merged**